src/core/Track.cpp

```diff
@@ -1070,12 +1114,47 @@ void TrackContentObjectView::mouseReleaseEvent( QMouseEvent * me )
 	{
 		setSelected( !isSelected() );
 	}
-
-	if( m_action == Move || m_action == Resize || m_action == ResizeLeft )
+	else if( m_action == Move || m_action == Resize || m_action == ResizeLeft )
 	{
 		// TODO: Fix m_tco->setJournalling() consistency
 		m_tco->setJournalling( true );
 	}
+	else if( m_action == Split )
+	{
+		SampleTCO * leftTCO = dynamic_cast<SampleTCO*>( m_tco );
+
+		if (leftTCO)
+		{
+			leftTCO->setMarkerEnabled( false );
+
+			const int relativePixelPos = me->pos().x();
+			const float ppt = m_trackView->trackContainerView()->pixelsPerTact();
+			MidiTime splitPos = relativePixelPos * MidiTime::ticksPerTact() / ppt;
+
+			if ( me->modifiers() & Qt::ControlModifier
+			  || me->modifiers() & Qt::AltModifier    ){}
+			else { splitPos = quantizeMarkerPos( splitPos, me->modifiers() & Qt::ShiftModifier ); }
```

Comment:
I would suggest:
```
if (!(me->modifiers() & Qt::ControlModifier || me->modifiers() & Qt::AltModifier))
{
    splitPos = quantizeMarkerPos(splitPos, me->modifiers() & Qt::ShiftModifier);
}
```

Reply:
Using the de Morgan's law, @BaraMGB's solution can be written as:
```cpp
if (!(me->modifiers() & Qt::ControlModifier)
 && !(me->modifiers() & Qt::AltModifier))
{
    splitPos = quantizeMarkerPos(splitPos, me->modifiers() & Qt::ShiftModifier);
}
```
I'm not sure if it impropves readability though.

Reply:
Just to throw another idea out there...
```cpp
if (!(me->modifiers() & (Qt::ControlModifier | Qt::AltModifier)))
```

Reply:
In the end I went with creating a function for this, which shortens the condition and makes it clear what's being checked for.